change/@ni-nimble-components-e9fe99af-bb91-40c3-80e3-f5b41e2b0ff6.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```

Comment:
FYI @jattasNI @m-akinc working on some tweaks in this branch: https://github.com/ni/nimble/tree/numeric-column-units-structure-tweaks

I'll handle merge from main, the open comments in the PR, and addressing some TODOs I added before merging that branch into this PR for a final lookover

Reply:
I haven't looked at this PR since the December refactors. I'll take another look once Milan finishes those changes.

Reply:
@m-akinc @jattasNI ready for review. I primarily reviewed the utilties/unit-format code, I reset myself to look at the element / table integration.

Major changes:
- Moved the related high-level overview docs to utilities/uunit-format/README.md. Would recommend reading iit to get a lay of the land for reviewing.
- Renamed the top-level classes to *UnitFormat, i.e. DefaultUnitFormat and DecimalUnitFormat. This was to more closely align with the Intl.NumberFormat/DateFormat/ListFormat/DataTimeFormat convention
- Updated the API surface to align with the Intl.NumberFormat API (use the same parameter names and defaults)
- Re-organized the layout to emphasize and split UnitScales and their associated objects from ManuallyTranslatedUnitScales and made naming align
- Generally switched to concrete classes and subclassing over interfaces where possible. Helped understand the different objects and responsibilities better with consistent naming.